packages/sdk-metrics/src/state/TemporalMetricProcessor.ts

```diff
@@ -134,7 +134,7 @@ export class TemporalMetricProcessor<T extends Maybe<Accumulation>> {
     });
 
     return this._aggregator.toMetricData(
-      instrumentDescriptor,
```

Comment:
We should be able to apply `toExternal` in the constructor of `TemporalMetricProcessor` and cache the result (the descriptor will not be modified anyways) to avoid `toExternal` calls in every aggregator.

Reply:
I think this change may require a large refactoring of the existing code, as the descriptor's `description` may change after the creation of the `MetricStorage`. :grimacing: This means that we'd have to propagate any changes to the description to the `TemporalMetricsProcessor` via the `MetricsStorage`, which then in turn means that we'll have to pull in the `TemporalMetricsProcessor` into the base class, and update it there, which then requires refactoring a quite a few tests to make it fit. :slightly_frowning_face: 

It looked quite simple at first but it adds a lot of complexity. Not sure if it is still worth changing. WDYT? :thinking:

Reply:
Now that `toExternal` is merely a type cast, what do you think about applying the type case here and avoiding repeating `toExternal` in every aggregator?